File: Java/58. Length of Last Word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int r = s.length()-1, l;
        while (r >= 0 && s.charAt(r) == ' ') {
             r--;
        }
        
        l = r-1;
        while (l >= 0 && s.charAt(l) != ' ') {
            l--;
        }
        
        return r-l;
    }
}

File: Python3/58. Length of Last Word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        for i in range(len(s)-1, -1, -1):
            if s[i] != ' ':
                l = i-1
                while l >= 0 and s[l] != ' ':
                    l -= 1
                return i - l
                    